stdlib/maptypes.go

```diff
@@ -0,0 +1,58 @@
+package stdlib
+
+import (
+	"encoding"
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+	"log"
+	"reflect"
+)
+
+func init() {
+	mt := []reflect.Type{
+		reflect.TypeOf((*fmt.Formatter)(nil)).Elem(),
+		reflect.TypeOf((*fmt.Stringer)(nil)).Elem(),
+	}
```

Comment:
What about adding also `fmt.GoStringer` ?

See https://pkg.go.dev/fmt#GoStringer

Reply:
In fact, the solution here is incomplete for `Printf`, as the non empty interface to target also depends in this case from the the format verbs, as `%#v` which requires a `GoStringer` instead of a `Stringer` for `%v`. For now we ignore this, and thus not adding `GoStringer`.

Having a full solution here for `Printf` would probably require to rewrite it inside the interpreter to parse the verbs in format string. It's too much of an edge case for us, and the current solution seems a better compromise.